src/layers/core/polygon-layer/polygon-layer.js

```diff
@@ -73,14 +73,14 @@ export default class PolygonLayer extends CompositeLayer {
     }
   }
 
-  _onHoverSubLayer(info) {
-    info.object = (info.object && info.object.feature) || info.object;
-    this.props.onHover(info);
-  }
+  _onPickSubLayer(handler, info) {
+    Object.assign(info, {
+      layer: this,
+      // override object with picked feature
+      object: (info.object && info.object.feature) || info.object
```

Comment:
1) My off-the-cuff idea is that `PolygonLayer` shouldn't know anything about `object.feature` (that being a `GeoJsonLayer` thing). Can we remove the reference?
2) If so, it would be cool if there was a way to provide some generic forwarding of picking callbacks, perhaps on the `CompositeLayer` class. i.e any reason why these couldn't be moved to that layer?